src/instruction/instructions/store.ts

```diff
@@ -0,0 +1,302 @@
+import { Halfword, Word } from 'types/binary'
+import {
+  checkOptionCount,
+  create,
+  createImmediateBits,
+  createLowRegisterBits,
+  createRegisterBits,
+  getBits,
+  isImmediate,
+  isOptionCountValid, registerStringHasBrackets, removeBracketsFromRegisterString,
+  setBits
+} from 'instruction/opcode'
+import { Registers } from 'board/registers'
+import { IMemory } from 'board/memory/interfaces'
+
+import { ILabelOffsets } from '../interfaces'
+import { BaseInstruction } from './baseInstruction'
+
+/**
+ * Represents a 'STORE' instruction - STR (immediate offset) - word
+ */
+export class StoreInstructionImmediateOffset extends BaseInstruction {
+  public name: string = 'STR'
+  public pattern: string = '01100XXXXXXXXXXX'
+  private rnPattern: string = '0110000000XXX000'
+  private rtPattern: string = '0110000000000XXX'
+  private immPattern: string = '01100XXXXX000000'
+  private expectedOptionCount: number = 3
+
+  public canEncodeInstruction(commandName: string, options: string[]): boolean {
+    if (super.canEncodeInstruction(commandName, options)) {
+      if (
+          isOptionCountValid(options, this.expectedOptionCount) &&
+          isImmediate(options[2]) &&
+          registerStringHasBrackets(options[1],options[2])
+      ) {
+        return true
+      }
+    }
+    return false
```

Comment:
can be converted to `return super.canEncode... && second if condition`

nice to have.

Reply:
Thank you for bringing this up. I just changed that.